src/dispatching/dialogue/storage/redis_storage.rs

```diff
@@ -51,36 +54,30 @@ where
 {
     type Error = RedisStorageError<<S as Serializer<D>>::Error>;
 
-    // `.del().ignore()` is much more readable than `.del()\n.ignore()`
-    #[rustfmt::skip]
     fn remove_dialogue(
         self: Arc<Self>,
         chat_id: i64,
     ) -> BoxFuture<'static, Result<(), Self::Error>> {
         Box::pin(async move {
-            let res = redis::pipe()
+            let deleted_rows_count = redis::pipe()
                 .atomic()
-                .get(chat_id)
-                .del(chat_id).ignore()
-                .query_async::<_, redis::Value>(
-                    self.conn.lock().await.deref_mut(),
-                )
+                .del(chat_id)
+                .query_async::<_, redis::Value>(self.conn.lock().await.deref_mut())
                 .await?;
-            // We're expecting `.pipe()` to return us an exactly one result in
-            // bulk, so all other branches should be unreachable
-            match res {
-                redis::Value::Bulk(bulk) if bulk.len() == 1 => {
-                    Option::<Vec<u8>>::from_redis_value(&bulk[0])?
-                        .map(|v| {
-                            self.serializer
-                                .deserialize(&v)
-                                .map_err(RedisStorageError::SerdeError)
-                        })
-                        .transpose()?;
-                    Ok(())
-                }
-                _ => unreachable!(),
+
+            let deleted_rows_count = match deleted_rows_count {
+                redis::Value::Bulk(values) => match values[0] {
+                    redis::Value::Int(x) => x,
+                    _ => unreachable!("Must return redis::Value::Int"),
+                },
+                _ => unreachable!("Must return redis::Value::Bulk"),
+            };
+
+            if deleted_rows_count == 0 {
+                return Err(RedisStorageError::RowNotFound);
             }
+
+            Ok(())
```

Comment:
Maybe this is better?
```suggestion
            if let redis::Value::Bulk(values) = deleted_rows_count {
                 if let redis::Value::Int(deleted_rows_count) = values[0] {
                     match deleted_rows_count {
                         0 => Err(RedisStorageError::RowNotFound),
                         _ => Ok(())
                     }
                 }
             }
                 
             unreachable!("Must return redis::Value::Bulk(redis::Value::Int(_))");
```